File: Serene/Serene.Script/Imports/ServerImports/Northwind.OrderDetailForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Northwind
{
    public partial class OrderDetailForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.OrderDetail";

        public OrderDetailForm(string idPrefix) : base(idPrefix) {}

        public LookupEditor ProductID { [InlineCode("{this}.w('ProductID', Serenity.LookupEditor)")] get; private set; }
        public DecimalEditor UnitPrice { [InlineCode("{this}.w('UnitPrice', Serenity.DecimalEditor)")] get; private set; }
        public IntegerEditor Quantity { [InlineCode("{this}.w('Quantity', Serenity.IntegerEditor)")] get; private set; }
        public DecimalEditor Discount { [InlineCode("{this}.w('Discount', Serenity.DecimalEditor)")] get; private set; }
    }
}

File: Serene/Serene.Core/Imports/ServerTypings/Northwind.OrderDetailForm.ts
namespace Serene.Northwind {
    export interface OrderDetailForm {
        ProductID: Serenity.LookupEditor;
        UnitPrice: Serenity.DecimalEditor;
        Quantity: Serenity.IntegerEditor;
        Discount: Serenity.DecimalEditor;
    }

    export class OrderDetailForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.OrderDetail';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!OrderDetailForm.init)  {
                OrderDetailForm.init = true;

                var s = Serenity;
                var w0 = s.LookupEditor;
                var w1 = s.DecimalEditor;
                var w2 = s.IntegerEditor;

                Q.initFormType(OrderDetailForm, [
                    'ProductID', w0,
                    'UnitPrice', w1,
                    'Quantity', w2,
                    'Discount', w1
                ]);
            }
        }
    }
}
